Enable 3 finger tap & 5 finger tap in multitouch mode to toggle screen keyboard & show quick menu

dangnm opened the pull request:
- Enable three-finger tap to show the on-screen keyboard in multi-touch mode: In the original Moonlight app, I frequently used this feature. However, it appears to be missing in multi-touch mode, and I'm not sure why it was removed. Re-enabling it would improve usability for those who rely on quick keyboard access.

- Enable five-finger tap to open the back/quick menu in multi-touch mode: Currently, the "swipe from the right edge" gesture conflicts with the right-side widget opening event when I'm using my tablet to remote into Windows. As a result, I have to swipe twice to access the quick menu. Adding a five-finger tap gesture would provide a more reliable alternative, especially for devices where edge swipes are not smooth or responsive.

dangnm committed:
enable 3 finger tap & 5 finger tap

enable 3 finger tap for showing on screen keyboard
enable 5 finger tap for showing back menu(quick menu)

ClassicOldSong commented:
It's not removed, it's just conflicting. That means you'll loose the ability to do 3 finger tap and 5 finger tap to the host.

May you also add an option to toggle 3/4 finger keyboard showing when multi touch is enabled? Or a more proper way, 5 finger back gesture is enabled by default but also disabled in multi touch mode, you can toggle all multi touch gestures instead of only the 5 finger one.

dangnm committed:
remove unsued code

enable 3 finger tap for showing on screen keyboard
enable 5 finger tap for showing back menu(quick menu)

ClassicOldSong commented:
Also, since you have separate out the multi touch gesture handling logic, may you also update the other non-multi touch parts to reuse the logic? It also seems 4 finger gesture bringing up the full keyboard is missing.

dangnm committed:
update logics for 4 fingers full keyboard

enable 3 finger tap for showing on screen keyboard
enable 5 finger tap for showing back menu(quick menu)

dangnm commented:
@ClassicOldSong I have updated the code & tested on my phone & my tablet. Thank you for your feedbacks.

ClassicOldSong commented:
Great, I think I still have to do some modifications myself though.

Thanks for your contribution!
